docs/how-to-guide/how-to-guide.md

```diff
@@ -0,0 +1,358 @@
+
+```
+echo <nginx-ingress-integrator-service-ip> <desired-hostname> | sudo tee -a /etc/hosts
+```
+
+Test the ingress by sending a GET request to the hostname.
+
+```
+curl <desired-hostname>
+```
+
+### COS integrations
+
+#### prometheus-k8s
+
+Deploy and relate [prometheus-k8s](https://charmhub.io/prometheus-k8s) charm with wordpress-k8s
+charm through the metrics-endpoint relation via prometheus_scrape interface. Prometheus should
+start scraping the metrics exposed at `:9117/metrics` endpoint.
+
+```
+juju deploy prometheus-k8s
+juju relate wordpress-k8s prometheus-k8s
+```
+
+#### loki-k8s
+
+Deploy and relate [loki-k8s](https://charmhub.io/loki-k8s) charm with wordpress-k8s charm through
+the logging relation via loki_push_api interface. Promtail worker should spawn and start pushing
+Apache access logs and error logs to loki.
+
+```
+juju deploy loki-k8s
+juju relate wordpress-k8s loki-k8s
+```
+
+#### grafana-k8s
+
+In order for the Grafana dashboard to function properly, grafana should be able to connect to
+Prometheus and Loki as its datasource. Deploy and relate prometheus-k8s and lok-k8s charm with
+[grafana-k8s](https://charmhub.io/grafana-k8s) charm through grafana-source relation.
+
+Note that the relation “grafana-source” has to be explicitly stated since prometheus-k8s and
+grafana-k8s share multiple interfaces.
+
+```
+juju deploy grafana-k8s
+juju relate prometheus-k8s:grafana-source grafana-k8s:grafana-source
+juju relate loki-k8s:grafana-source grafana-k8s:grafana-source
+```
+
+Then, wordpress-k8s charm can be related with grafana using the grafana-dashboard relation with
+grafana_dashboard interface.
+
+```
+juju relate wordpress-k8s grafana-k8s
+```
+
+To access the Grafana dashboard for wordpress-k8s charm, run the following command to obtain
+credentials for admin access.
+
+```
+juju run-action grafana-k8s/0 get-admin-password --wait
+```
+
+You can now log into the grafana dashboard by visiting `http://<grafana-unit-ip>:3000`. Navigate to
+`http://<grafana-unit-ip>:3000/dashboards` and access the WordPress dashboard named Wordpress Operator
+Overview.
+
+### ObjectStorage with Swift
+
+Follow the instructions on installing OpenStack from the OpenStack
+[documentation](https://docs.openstack.org/install-guide/). For testing purposes, you can install
+[DevStack](https://docs.openstack.org/devstack/latest/).
+
+After successful installations, you should be able to see `openrc` file at the location of
+installation. The contents of the `openrc` file should look something similar to the following.
+
+```
+export OS_REGION_NAME=RegionOne
+export OS_PROJECT_DOMAIN_ID=default
+export OS_CACERT=
+export OS_TENANT_NAME=demo
+export OS_USER_DOMAIN_ID=default
+export OS_USERNAME=demo
+export OS_VOLUME_API_VERSION=3
+export OS_AUTH_TYPE=<strong-password>
+export OS_PROJECT_NAME=demo
+export OS_PASSWORD=<strong-password>
+export OS_IDENTITY_API_VERSION=3
+export OS_AUTH_URL=http://<openstack-deployment-address>/identity
+```
```

Comment:
If you install the DevStack in your development machine, the `openrc` file is not like the code snippet above, it will be something like this. 

```
#!/usr/bin/env bash                                                                                                                                                           [65/206]
#                                                                                                                                                                                     
# source openrc [username] [projectname]                                                                                                                                              
#                                                                                                                                                                                     
# Configure a set of credentials for $PROJECT/$USERNAME:                                                                                                                              
#   Set OS_PROJECT_NAME to override the default project 'demo'                             
#   Set OS_USERNAME to override the default user name 'demo'
#   Set ADMIN_PASSWORD to set the password for 'admin' and 'demo'

# NOTE: support for the old NOVA_* novaclient environment variables has
# been removed.

if [[ -n "$1" ]]; then
    OS_USERNAME=$1
fi
if [[ -n "$2" ]]; then
    OS_PROJECT_NAME=$2
fi

# Find the other rc files
RC_DIR=$(cd $(dirname "${BASH_SOURCE:-$0}") && pwd)

# Import common functions
source $RC_DIR/functions

# Load local configuration
source $RC_DIR/stackrc

# Load the last env variables if available
if [[ -r $RC_DIR/.stackenv ]]; then
    source $RC_DIR/.stackenv
    export OS_CACERT
fi

# Get some necessary configuration
source $RC_DIR/lib/tls

# The OpenStack ecosystem has standardized the term **project** as the
# entity that owns resources.  In some places **tenant** remains
# referenced, but in all cases this just means **project**.  We will
# warn if we need to turn on legacy **tenant** support to have a
# working environment.
export OS_PROJECT_NAME=${OS_PROJECT_NAME:-demo}

echo "WARNING: setting legacy OS_TENANT_NAME to support cli tools."
export OS_TENANT_NAME=$OS_PROJECT_NAME

# In addition to the owning entity (project), nova stores the entity performing
# the action as the **user**.

...
```

But I think you can get a list of credentials using this command:

```bash
source openrc && printenv | grep OS_
```

Reply:
Replaced the example to match the output of Devstack setup.
Thank you for the suggestion :)